### ISSUE TITLE
[RFC] New Module Validator API

### ISSUE THREAD

**Issue by Darktex** [pytorch/opacus#133]
## 🚀 Feature
Not all `nn.Module`s can compute a `.grad_sample`, and of those that can, not all can be used with DP-SGD (BatchNorm can't). We should have a component that owns checking and fixing complex models so that they can be easily used with Opacus.

We need one (or more) components to do the following:
1. **Validate** that your model can be trained by Opacus
2. **Fix** it whenever possible (for example, replacing `nn.LSTM` with `opacus.DPLSTM` or replacing `nn.BatchNorm` with `nn.GroupNorm`). 

This RFC proposal proposes a component, `opacus.Validator` to do both of the above.

## Pitch

This proposal is composed of 3 items:

1. Design doc: [[Public RFC] Module Validator API](https://docs.google.com/document/d/1SwfaOTcfca7R2ZouoTNBlVKG5d722QDZ4tUAoFtAIXg/edit?usp=sharing)
2. A working(ish) PR: #131 
3. An example Colab on what the API currently looks like (based on the PR so you can play with it): https://colab.research.google.com/drive/1rQFSVUoKJeSQ0ht8oIBscSu-SW3brraE?usp=sharing
